server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/controllers/AdmissionController.java

```diff
@@ -54,17 +75,31 @@ public String getName() {
     }
 
     /**
-     * Adds the rejection count for the controller. Primarily used when copying controller states.
-     * @param count To add the value of the tracking resource object as the provided count
+     * Add rejection count to the rejection count metric tracked by the admission controller
      */
-    public void addRejectionCount(long count) {
-        this.rejectionCount.addAndGet(count);
+    public void addRejectionCount(String admissionControlActionType, long count) {
```

Comment:
This will always allocate AutomicLong on heap and this can be very costly on GC and might not be good memory efficient. Can be changes like below
```
        if (!this.rejectionCountMap.containsKey(admissionControlActionType)) {
            this.rejectionCountMap.put(admissionControlActionType, new AtomicLong(0));
        }
       this.rejectionCountMap.get(admissionControlActionType).getAndAdd(count);
```

The current code is not thread safe as well,.....

Reply:
Also aren't we adding by `1` always whenever any rejection happens ? We can hardcode `increamentAndGet` in that case.

Reply:
Yes will get this updated.

> Also aren't we adding by 1 always whenever any rejection happens ? We can hardcode increamentAndGet in that case.

Yes we are adding 1 for every rejection. Just to make this method as generic so passing the count as parameter.